sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -0,0 +1,735 @@
+                            break;
+                        case TypedConstantKind.Type:
+                            break;
+                        case TypedConstantKind.Array:
+                            var arrayValues = arg.Values.Select(a => a.Value!.ToString()).ToArray();
+                            dict[argumentName] = arrayValues;
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                return true;
+            }
+
+            /// <summary>
+            /// This method verifies that an EventHubsTrigger matches our expectations on cardinality (isBatched property). If isBatched is set to true, the parameter with the
+            /// attriute must be an enumerabletype.
+            /// </summary>
+            private bool IsEventHubsTriggerValid(IParameterSymbol parameterSymbol, TypeSyntax? parameterTypeSyntax, SemanticModel model, AttributeData attribute, out DataType dataType, out Cardinality cardinality)
+            {
+                dataType = DataType.Undefined;
+                cardinality = Cardinality.Undefined;
+
+                // Check if IsBatched is false (by default it is true and does not appear in the attribute constructor)
+                foreach (var arg in attribute.NamedArguments)
+                {
+                    if (String.Equals(arg.Key, Constants.IsBatchedKey) &&
+                        arg.Value.Value != null)
+                    {
+                        var isBatched = (bool)arg.Value.Value; // isBatched takes in booleans so we can just type cast it here to use
+
+                        if (!isBatched)
+                        {
+                            dataType = GetDataType(parameterSymbol.Type);
+                            cardinality = Cardinality.One;
+                            return true;
+                        }
+                    }
+                }
+
+                // we check if the param is an array type
+                // we exclude byte arrays (byte[]) b/c we handle that as cardinality one (we handle this simliar to how a char[] is basically a string)
+                if (parameterSymbol.Type is IArrayTypeSymbol && !SymbolEqualityComparer.Default.Equals(parameterSymbol.Type, Compilation.GetTypeByMetadataName(Constants.ByteArrayType)))
+                {
+                    dataType = GetDataType(parameterSymbol.Type);
+                    cardinality = Cardinality.Many;
+                    return true;
+                }
+
+                var isGenericEnumerable = parameterSymbol.Type.IsOrImplementsOrDerivesFrom(Compilation.GetTypeByMetadataName(Constants.IEnumerableGenericType));
+                var isEnumerable = parameterSymbol.Type.IsOrImplementsOrDerivesFrom(Compilation.GetTypeByMetadataName(Constants.IEnumerableType));
+
+                // Check if mapping type
```

Comment:
Did not quite understand this. Could you add a comment to code to help future readers?